packages/gestalt/src/ButtonLink.js

```diff
@@ -0,0 +1,200 @@
+     See [GlobalEventsHandlerProvider](/web/utilities/globaleventshandlerprovider#Link-handlers) to learn more about link navigation.
+   */
+  onClick?: ({|
+    event:
+      | SyntheticMouseEvent<HTMLButtonElement>
+      | SyntheticMouseEvent<HTMLAnchorElement>
+      | SyntheticKeyboardEvent<HTMLAnchorElement>
+      | SyntheticKeyboardEvent<HTMLButtonElement>,
+    dangerouslyDisableOnNavigation: () => void,
+  |}) => void,
+  /**
+   * sm: 32px, md: 40px, lg: 48px
+   */
+  size?: 'sm' | 'md' | 'lg',
+  /**
+   * Text to render inside the Button to convey the function and purpose of the Button.
+   */
+  text: string,
+  /**
+   * Specifies a link URL.
+   */
+  href: string,
+  /**
+   * Provides hints for SEO. See the [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#rel) to learn more.
+   */
+  rel?: 'none' | 'nofollow',
+  /**
+   * Indicates the browsing context where an href will be opened.
+   */
+  target?: Target,
+|};
+
+const ButtonLinkWithForwardRef: AbstractComponent<ButtonType, HTMLAnchorElement> = forwardRef<
+  ButtonType,
+  HTMLAnchorElement,
+>(function ButtonLink(props: ButtonType, ref): Node {
+  const {
+    accessibilityLabel,
+    color = 'gray',
+    dataTestId,
+    disabled = false,
+    fullWidth = false,
+    iconEnd,
+    onClick,
+    tabIndex = 0,
+    size = 'md',
+    text,
+  } = props;
+
+  const innerRef = useRef<null | HTMLAnchorElement>(null);
+
+  // When using both forwardRef and innerRef, React.useimperativehandle() allows a parent component
+  // that renders <ButtonLink ref={inputRef} /> to call inputRef.current.focus()
+  useImperativeHandle(ref, () => innerRef.current);
+
+  const { accessibilityNewTabLabel } = useDefaultLabelContext('Link');
+
+  const { name: colorSchemeName } = useColorScheme();
+  // We need to make a few exceptions for accessibility reasons in darkMode for red buttons
+  const isDarkMode = colorSchemeName === 'darkMode';
+  const isDarkModeRed = isDarkMode && color === 'red';
+  const isDarkModeBlue = isDarkMode && color === 'blue';
+
+  let colorClass = color === 'transparentWhiteText' ? 'transparent' : color;
+  if (isDarkModeRed) {
+    colorClass = 'darkModeRed';
+  }
+  const textColor =
+    (disabled && 'subtle') ||
+    'inverse' ||
+    ((isDarkModeRed || isDarkModeBlue) && 'default') ||
+    DEFAULT_TEXT_COLORS[color];
+  const { href, rel = 'none', target = null } = props;
+  const ariaLabel = getAriaLabel({ target, accessibilityLabel, accessibilityNewTabLabel });
+
+  const handleClick = (
+    event: SyntheticKeyboardEvent<HTMLAnchorElement> | SyntheticMouseEvent<HTMLAnchorElement>,
+    dangerouslyDisableOnNavigation: () => void,
+  ) =>
+    onClick
+      ? onClick({
+          event,
+          dangerouslyDisableOnNavigation: dangerouslyDisableOnNavigation ?? (() => {}),
+        })
+      : undefined;
+
+  const handleLinkClick = ({
```

Comment:
can we merge handleLinkClick & handleClick before they were used sepaartrely not anymore in this component

Reply:
DONE